src/strategies/FluentStrategyMixins.ts

```diff
@@ -0,0 +1,55 @@
+import {Arbitrary, FluentPick} from '../arbitraries'
+import {FluentResult} from '../FluentCheck'
+import {FluentStrategy, FluentStrategyInterface} from './FluentStrategy'
+
+type MixinConstructor<T = {}> = new (...args: any[]) => T
+type MixinStrategy = MixinConstructor<FluentStrategy>
+
+export function Random<TBase extends MixinStrategy>(Base: TBase) {
+  return class extends Base implements FluentStrategyInterface {
+    hasInput<K extends string>(arbitraryName: K): boolean {
+      return this.arbitraries[arbitraryName] !== undefined &&
+        this.arbitraries[arbitraryName].pickNum < this.arbitraries[arbitraryName].collection.length
+    }
+
+    getInput<K extends string, A>(arbitraryName: K): FluentPick<A> {
+      return this.arbitraries[arbitraryName].collection[this.arbitraries[arbitraryName].pickNum++]
+    }
+
+    handleResult() {}
+  }
+}
+
+export function Shrinkable<TBase extends MixinStrategy>(Base: TBase) {
+  return class extends Base {
+    shrink<K extends string>(arbitraryName: K, partial: FluentResult | undefined, depth: number) {
+      const shrinkedArbitrary = this.arbitraries[arbitraryName].arbitrary.shrink(partial!.example[arbitraryName])
+      this.arbitraries[arbitraryName].collection = this.buildArbitraryCollection(shrinkedArbitrary,
+        this.configuration.shrinkSize)
+    }
+  }
+}
+
+export function Dedupable<TBase extends MixinStrategy>(Base: TBase) {
```

Comment:
Is this still relevant?

Reply:
It is in terms of deciding whether sampling must be unique or not. Maybe we should rename this for something else. Do you have any suggestions?

Reply:
I was asking because I thought sampling should always be unique. But I remembered that we had mentioned that in some cases, removing uniques might be too expensive (e.g., space-wise).

Reply:
Oh ok, I misunderstood. Can I consider this conversation finished then?

Reply:
Yes